Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -162,6 +175,14 @@ build pkg_descr lbi flags suffixes = do
     distPref = fromFlag (buildDistPref flags)
     verbosity = fromFlag (buildVerbosity flags)
 
+checkBuildProblems
+  :: Verbosity -> Compiler -> BuildFlags -> IO ()
+checkBuildProblems verbosity comp flags = do
+  unless (jsemSupported comp || not (isJust (flagToMaybe (buildUseSemaphore flags)))) $
```

Comment:
What is the second condition? Not having --semaphore? It seems not explained in the error message.

Reply:
Basically this translates to "Unless the compiler supports semaphores, or the semaphore flag has not been passed, throw an error".

Reply:
correct, I would have thought a `when` expressed the warning in a better way, but that is fine.